lstchain/high_level/hdu_table.py

```diff
@@ -195,28 +195,19 @@ def create_hdu_index_hdu(file_list, hdu_index_file, overwrite=False):
         t_pnt["HDU_NAME"] = "POINTING"
 
         hdu_index_tables.append(t_pnt)
-        hdu_names = [
-            "EFFECTIVE AREA",
-            "ENERGY DISPERSION",
-            "BACKGROUND",
-            "PSF",
-            "RAD_MAX",
-        ]
-
-        for irf in hdu_names:
-            try:
-                t_irf = t_events.copy()
-                irf_hdu = hdu_list[irf].header["HDUCLAS4"]
-
-                t_irf["HDU_CLASS"] = irf_hdu.lower()
-                t_irf["HDU_TYPE"] = irf_hdu.lower().strip(
-                    "_" + irf_hdu.lower().split("_")[-1]
-                )
-                t_irf["HDU_NAME"] = irf
-                hdu_index_tables.append(t_irf)
-            except KeyError:
-                log.error(f"Run {t_events['OBS_ID']} does not contain HDU {irf}")
 
+        # 0:PRIMARY, 1:EVENTS, 2:GTI, 3:POINTING, 4-:IRF 
+        for hdu in hdu_list[4:]:
+            irf_hdu = hdu.header["HDUCLAS4"]
```

Comment:
This will not work for the [GH_CUTS](https://github.com/cta-observatory/cta-lstchain/blob/main/lstchain/tools/lstchain_create_irf_files.py#L674-L687) and [AL_CUTS](https://github.com/cta-observatory/cta-lstchain/blob/main/lstchain/tools/lstchain_create_irf_files.py#L701-L714) HDU as they are not pyirf created HDUs and do not (and should not) have HDUCLAS4 header.

Reply:
It would be better to include GH_CUTS and AL_CUTS as well? The original code didn't include those cuts in `hdu_names`

Reply:
okay I see! Maybe we can keep it as it was in this PR. Then let's make another PR if needed!

Reply:
This then checks the HDUs `GH_CUTS` and `AL_CUTS` for `HDUCLAS4`, where it is not present

Reply:
The tests pass because there is only a single test for creating the DL3 index files, which uses the source-independent DL3 files with global cuts.
Maybe adding a test for indexing the source-dependent DL3 files with energy-dependent cuts would be a good measure. (Basically replicating the current test for the path `temp_dir_observed_srcdep_files` instead of `temp_dir_observed_files`)